spring-projects-experimental/spring-authorization-server
# Authorization Endpoint filter for Authorization Code flow

Fixes #66 

Functionality Covered
-------------------------
- Validation of client id & response type
- Generate authorization code
- Save Authorization
- Redirect to client redirect uri
- Send error in http response
- Send error as query params in redirect uri

Validations covered
-------------------
- Client Id should be mandatory
- Client Id should be registered
- Client Id should be configured with Authorization grant type
- Response type is mandaotry
- Response type should be code
- Redirect Uri should be specified if client has multiple redirect uris configured
- If supplied, redirect uri should be same as the one configured on client
- User should be authenticated
- Endpoint should be /oauth2/authorize
